src/parser/LiteralOrIriType.h

```diff
@@ -0,0 +1,50 @@
+// Copyright 2023, University of Freiburg,
+//                 Chair of Algorithms and Data Structures.
+// Author: Benedikt Maria Beckermann <benedikt.beckermann@dagstuhl.de>
+
+#pragma once
+
+#include "IriType.h"
+#include "LiteralType.h"
+
+class LiteralOrIriType {
+ private:
+  using LiteralOrIriVariant = std::variant<LiteralType, IriType>;
+  LiteralOrIriVariant data;
+
+  // Returns contained IriType object if available, throws exception otherwise
+  IriType& getIriTypeObject();
+  // Returns contained LiteralType object if available, throws exception
+  // otherwise
+  LiteralType& getLiteralTypeObject();
+
+ public:
+  // Creates a new LiteralOrIriType based on a LiteralType object
+  explicit LiteralOrIriType(LiteralType data);
+  // Creates a new LiteralOrIriType based on a IriType object
+  explicit LiteralOrIriType(IriType data);
+
+  // Returns true, if object contains an IriType object
+  [[nodiscard]] bool isIri() const;
+  // Returns iri string of contained IriType object if available, throws
+  // exception otherwise
+  NormalizedStringView getIriString();
+
+  // Returns true, if object contains an LiteralType object
+  [[nodiscard]] bool isLiteral() const;
```

Comment:
Let me guess, you are using Clion.
That "every const Method should be `[[nodiscard]]`" rule of Clang-Tidy is much too aggressive as you typically don't gain much (the bugs you introduce that way are also found otherwise).
You can deactivate this check if the yellow stuff annoys you.
Use `[[nodiscard]]` only when ignoring the return value leads to a very very subtlle bug (e.g. for scope guards and other RAII types).